### PR TITLE
Camel 13760

### PR DESCRIPTION
**Pull request by davsclaus**
CAMEL-13760: camel3 - Remove configuring prefix and suffix token on property placeholder as it clashes and causes problems such as with simple language, and potentially elsewhere.

https://issues.apache.org/jira/browse/CAMEL-13760

Changing those prefix/suffix dont 100% work in all cases, and most users are using the default tokens. We should not allow to change them anymore. So they are removed for Camel 3.

### CODE REVIEW

File: components/camel-properties/src/main/java/org/apache/camel/component/properties/PropertiesComponent.java [hunk 1]
```diff
@@ -350,36 +346,12 @@ public void setIgnoreMissingLocation(boolean ignoreMissingLocation) {
...
-        if (prefixToken == null) {
-            this.prefixToken = DEFAULT_PREFIX_TOKEN;
-        } else {
-            this.prefixToken = prefixToken;
-        }
+        return PREFIX_TOKEN;
     }
 
     @ManagedAttribute(description = "Suffix token")
     public String getSuffixToken() {
```
Comment (bedlaj) [inline components/camel-properties/src/main/java/org/apache/camel/component/properties/PropertiesComponent.java]: getPrefixToken and getSuffixToken is removed from from SPI interface, but kept in implementation. Is this intention? These two methods in impl are unused, just exposing JMX and this is IMHO not needed to be in JMX now, when it is constant.

File: core/camel-api/src/main/java/org/apache/camel/spi/PropertiesComponent.java [hunk 2]
```diff
@@ -30,32 +30,20 @@
...
-     * The value of the prefix token used to identify properties to replace.
-     * Is default {@link #DEFAULT_PREFIX_TOKEN}
-     */
-    String getPrefixToken();
-
-    /**
-     * The value of the suffix token used to identify properties to replace.
-     * Is default {@link #DEFAULT_SUFFIX_TOKEN}
-     */
-    String getSuffixToken();
```
Comment (bedlaj) [inline core/camel-api/src/main/java/org/apache/camel/spi/PropertiesComponent.java]: AMQPConnectionDetails class is using these methods.

**Comment by davsclaus** [comment]
Thanks for the review Jan

**Comment by davsclaus** [comment]
@lburgazzoli I wonder if you recall something that camel-quarkus or camel-k may do to use different prefix tokens for property placeholders - I think I have seen `[[ ]]` used somewhere.

With this we dont allow that anymore, and use one standard which is `{{ }}` to make it all consistent and also avoid a few conditions where only `{{ }}` was really supported.

**Comment by lburgazzoli** [comment]
As far as I can recall, we do use `{{ }}` for both quarkus and camel-k, I remember we had `[[ ]]` for differentiate placeholders in blueprints but not directly related to camel

**Comment by davsclaus** [comment]
Thanks, then lets merge this

**Pull request merged**